apache/metron
# METRON-1997: Replace Threat Triage Score Field Slider with Text Box

## Contributor Comments

This PR replaces the score slider with an ace editor to highlight Stellar syntax. From now on users can set the score by adding a Stellar expression. It's backward compatible so the user still can add numbers as scores but they have to type it instead of setting the slider.

Since score has become a string and isn't a number anymore we cannot group or sort rules by that. I removed the two components which were responsible for sorting and grouping.

This is what I'm talking about:

![screen shot 2019-02-28 at 16 15 16](https://user-images.githubusercontent.com/2196208/53576558-43211f00-3b74-11e9-95e7-a465c833908f.png)

This PR also adds syntax highlighting to the `rule` field. The label "TEXT" was rewritten to "RULE" to make it clear what we're editing.

**A few things to keep in mind**

- If you add two or more rules with the exact same `rule` it just simply doesn't add it to the list (it should be more responsive by letting the user know)

**Testing**

1. Go to `/sensors`
2. Click on a sensor's edit button (the pen on the right)
3. Look for the `Threat Triage` section in the right floating pane
3. Click on the Threat Triage section's expander (it's the icon on the right)
4. In the second pane, click on the big `+` button at the bottom
5. In the 3rd panel add a name, a rule and a score (a score can be a Stellar expression like `value * 10`)
6. Click on the `Save` button. It'll validate the expression for you. If it's not, it'll let you know.
7. Check the newly added rule in the rule list in the 2nd pane
9. Take a quick look at the Threat Triage section in the 1st pane if it's ok
8. Don't forget to click on the Save button in the 1st pane

9. Open the read-only sensor pane (clicking on anywhere else but the pen on the row)
10. Find the Threat Triage section
11. Check if it's ok

![threat-triage](https://user-images.githubusercontent.com/2196208/53577608-3e5d6a80-3b76-11e9-8f8b-77af3454dbb4.gif)

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
